File: Editor/src/main/java/org/turbomedia/turboedit/editor/panes/timeline/TimelinePane.java
package org.turbomedia.turboedit.editor.panes.timeline;

import javafx.geometry.Insets;
import javafx.scene.layout.*;

public class TimelinePane extends GridPane {

    public TimelinePane() {
        double GRID_GAP = 6;

        var column = new ColumnConstraints();
        column.setHgrow(Priority.ALWAYS);

        var row = new RowConstraints();
        row.setVgrow(Priority.ALWAYS);

        getColumnConstraints().add(column);
        getRowConstraints().add(row);

        add(new UpperBar(), 0, 0, 1, 1);

        var videoLayer1 = new HBox(new LayerBox(LayerBox.LayerType.VIDEO), new ClipContainer(LayerBox.LayerType.VIDEO));
        videoLayer1.setSpacing(GRID_GAP);

        var audioLayer1 = new HBox(new LayerBox(LayerBox.LayerType.AUDIO), new ClipContainer(LayerBox.LayerType.AUDIO));
        audioLayer1.setSpacing(GRID_GAP);

        add(videoLayer1, 0, 1, 1, 1);
        add(audioLayer1, 0, 2, 1, 1);

        setHgap(GRID_GAP);
        setVgap(GRID_GAP);
        setPadding(new Insets(15));
    }
}

File: Editor/src/main/kotlin/turboedit/editor/panes/timeline/TimelinePane.kt
package turboedit.editor.panes.timeline

import javafx.geometry.Insets
import javafx.scene.layout.*

class TimelinePane : GridPane() {
    init {
        val gridGap = 6.0

        val column = ColumnConstraints()
        column.hgrow = Priority.ALWAYS

        val row = RowConstraints()
        row.vgrow = Priority.ALWAYS

        columnConstraints.add(column)
        rowConstraints.add(row)

        add(UpperBar(), 0, 0, 1, 1)

        val videoLayer1 = HBox(LayerBox(LayerBox.LayerType.VIDEO), ClipContainer(LayerBox.LayerType.VIDEO))
        videoLayer1.spacing = gridGap

        val audioLayer1 = HBox(LayerBox(LayerBox.LayerType.AUDIO), ClipContainer(LayerBox.LayerType.AUDIO))
        audioLayer1.spacing = gridGap

        add(videoLayer1, 0, 1, 1, 1)
        add(audioLayer1, 0, 2, 1, 1)

        hgap = gridGap
        vgap = gridGap
        padding = Insets(15.0)
    }
}
